eureka-client/src/test/java/com/netflix/discovery/util/EurekaEntityFunctionsTests.java

```diff
@@ -0,0 +1,359 @@
+package com.netflix.discovery.util;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.discovery.shared.Application;
+import com.netflix.discovery.shared.Applications;
+import org.junit.Assert;
+import org.junit.Test;
+import org.powermock.api.mockito.PowerMockito;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class EurekaEntityFunctionsTests {
+
+    @Test
+    public void testSelectApplicationNames() {
```

Comment:
Rename the test methods to describe the method tested, the condition and the expected result. Feel free to use your best judgement but it has to provide all of that info.

e.g. `testTakeFirstIfNonemptyShoudReturnFirst()`

throughout.

Reply:
test method names have been change